src/tagifyme/control/Controller.java

```diff
@@ -1,107 +1,148 @@
+            String pTName = "";
+            if (pT != null) {
+                theModel.addRelationship(new Relationship(d, pT));
+                pTName = pT.getName();
+            }
 
-                    theView.showAddDataDialogBox();
-                    ArrayList myStrings = new ArrayList<String>();
-                    int i = 0;
-                    for (Tag tags : theModel.tags() ){
-                        i++;
-                        myStrings.add(tags.getName());
-                    }
-                    String[] strings = new String[myStrings.size()];
-                    myStrings.toArray(strings);
-                    theView.populateTagList(strings);
-                    theModel.notifyObservers();
-                }
-                
-                if(command == "Delete Data"){
-                    theView.removeData("Test2");
-                }
-                
-                if(command == "Complete"){
-                    String Name = theView.getDataName();
-                    String PATH = theView.getDataPATH();
-                    String tagName = theView.getTagName();
-                    Data newData = new Data(Name, PATH);
-                    theModel.addData(newData);
-                    Tag correspondingTag = theModel.getTag(tagName);
-                    theModel.addRelationship(new Relationship(newData, correspondingTag));
-                    theView.addData(Name, PATH, tagName);
-                    theView.hideAddDataDialogBox();
-                }
-                
-                if (command == "Add Tag"){
-                    theView.showAddTagDialogBox();
-                }
-                
-                if (command == "ADD TAG"){
-                    String Name = theView.getNewTagName();
-                    theModel.addTag(new Tag(Name));
-                    theView.hideAddTagDialogBox();
-                }
+            // Propagate the changes to the model.
+            theView.addData(d.getName(), d.getPATH(), pTName);
+        }
 
-            } catch (Exception exm) {
-                System.out.println(exm.getMessage());
-            }
+        /**
+         * The User has selected some Data to delete, propagate
+         * that downward to the model.
+         */
+        private void handleDELETE_DATA() {
+            // TODO: It's likely that this should be renamed
+            // as a getter.
+            String nameToDelete = theView.deleteSelectedData();
+            theModel.deleteDataByPATH(nameToDelete);
+        }
 
+        /**
+         * Populate the UI with dialogs that allow for the
+         * addition of tags.
+         */
+        private void handleVIEW_ADD_TAG_DIALOG() {
+            theView.showAddTagDialogBox();
         }
 
-    }
+        /**
+         * Populate the UI with dialogs that allow for the
+         * deletion of Tags.
+         */
+        private void handleVIEW_DELETE_TAG_DIALOG() {
+            // TODO: What to populate the list with? I don't
```

Comment:
why

Reply:
I believe you were iterating over `Tag` and then populating the List via String; If I'm attempting to put GFX in the list and the GFX is built off the Tag, it might make more sense to just pass the Tag.

I threw that in there to keep it in my head between commits.